internal/ls/diagnostics.go

```diff
@@ -1,14 +1,15 @@
 package ls
 
 import (
+	"context"
 	"slices"
 
 	"github.com/microsoft/typescript-go/internal/ast"
 )
 
 func (l *LanguageService) GetDocumentDiagnostics(fileName string) []*ast.Diagnostic {
 	program, file := l.getProgramAndFile(fileName)
-	syntaxDiagnostics := program.GetSyntacticDiagnostics(file)
-	semanticDiagnostics := program.GetSemanticDiagnostics(file)
+	syntaxDiagnostics := program.GetSyntacticDiagnostics(context.Background(), file)
+	semanticDiagnostics := program.GetSemanticDiagnostics(context.Background(), file)
```

Comment:
```suggestion
	syntaxDiagnostics := program.GetSyntacticDiagnostics(context.TODO(), file)
	semanticDiagnostics := program.GetSemanticDiagnostics(context.TODO(), file)
```

I think this is one place we're going to want to plumb it.

Reply:
Indeed.

Reply:
I'm going to leave that to a separate PR though.

Reply:
To be clear my suggestion wasn't to plumb it in this PR, but to just leave these as TODO calls to make it clear that we're intending on changing it later.

Reply:
I'll approve it in any case, this is just a nit and I'm sure Andrew will figure this out when he plugs it in